server/registry.go

```diff
@@ -134,9 +134,6 @@ func createMultiMimeTypeFn(handlersByMimeType map[handlerDTOMimeTypeKey]*handler
 	available := lo.Map(values, func(hDTO *handlerDTO, _ int) contenttype.MediaType {
 		return hDTO.MediaType
 	})
-	availableConsumes := lo.Map(values, func(hDTO *handlerDTO, _ int) contenttype.MediaType {
```

Comment:
this availableConsumes array is created as a closure, as a result, each call to the handler will append a new entry to the slice and that never gets removed!